elephant_vending_machine/views.py

```diff
@@ -108,11 +131,16 @@ def upload_image():
             file.save(os.path.join(save_path, filename))
             response = "Success: Image saved."
             response_code = 201
+
+            try:
+                add_remote_image(save_path, filename)
+            except CalledProcessError:
+                response = "Error: Failed to copy file to hosts"
+                response_code = 400
```

Comment:
400s responses usually indicate an issue caused by the clients request. Since this would presumably be an internal error should we instead provide a 500? This would only be raised after the file has been validated and saved on the web server pi.

Reply:
Good call, 500 is probably the more appropriate response here.